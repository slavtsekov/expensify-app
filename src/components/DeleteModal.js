import React from "react";
import Modal from "react-modal";

const DeleteModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        onRequestClose={props.handleDecline}
        contentLabel="Selected Option"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">Delete</h3>
        <p className="modal__body">Are you sure you want to delete the selected expense?</p>
        <button className="button modal__confirm" onClick={props.handleAccept}>OK</button>
        <button className="button button--secondary" onClick={props.handleDecline}>Cancel</button>
    </Modal>
);

export { DeleteModal as default };