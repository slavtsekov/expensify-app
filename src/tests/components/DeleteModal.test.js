import React from "react";
import { shallow } from "enzyme";
import DeleteModal from "../../components/DeleteModal";

test("should render DeleteModal correctly", () => {
    const wrapper = shallow(<DeleteModal isOpen={true} handleAccept={()=>{}} handleDecline={()=>{}} />);

    expect(wrapper).toMatchSnapshot();
});

test("should handle primary button click", () => {
    const handleAcceptSpy = jest.fn();
    const wrapper = shallow(<DeleteModal isOpen={true} handleAccept={handleAcceptSpy} handleDecline={()=>{}} />);

    wrapper.find("button").at(0).simulate("click");

    expect(handleAcceptSpy).toHaveBeenCalled();
});

test("should handle secondary button click", () => {
    const handleDeclineSpy = jest.fn();
    const wrapper = shallow(<DeleteModal isOpen={true} handleAccept={()=>{}} handleDecline={handleDeclineSpy} />);

    wrapper.find("button").at(1).simulate("click");

    expect(handleDeclineSpy).toHaveBeenCalled();
});