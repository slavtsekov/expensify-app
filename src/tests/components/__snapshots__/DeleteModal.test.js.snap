// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeleteModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="Selected Option"
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
>
  <h3
    className="modal__title"
  >
    Delete
  </h3>
  <p
    className="modal__body"
  >
    Are you sure you want to delete the selected expense?
  </p>
  <button
    className="button modal__confirm"
    onClick={[Function]}
  >
    OK
  </button>
  <button
    className="button button--secondary"
    onClick={[Function]}
  >
    Cancel
  </button>
</Modal>
`;
